Source/Events.Processing/Projections/ProjectionKeys.cs

```diff
@@ -17,61 +20,89 @@ namespace Dolittle.Runtime.Events.Processing.Projections;
 public class ProjectionKeys : IProjectionKeys
 {
     readonly IProjectionKeyPropertyExtractor _keyPropertyExtractor;
+    readonly ILogger _logger;
 
     /// <summary>
     /// Initializes an instance of the <see cref="ProjectionKeys" /> class.
     /// </summary>
     /// <param name="keyPropertyExtractor">The projection key property extractor.</param>
-    public ProjectionKeys(IProjectionKeyPropertyExtractor keyPropertyExtractor)
+    /// <param name="logger">The <see cref="ILogger"/>.</param>
+    public ProjectionKeys(IProjectionKeyPropertyExtractor keyPropertyExtractor, ILogger logger)
     {
         _keyPropertyExtractor = keyPropertyExtractor;
+        _logger = logger;
     }
 
     public bool TryGetFor(ProjectionDefinition projectionDefinition, CommittedEvent @event, PartitionId partition, out ProjectionKey key)
     {
         key = null;
         var eventSelector = projectionDefinition.Events.FirstOrDefault(_ => _.EventType == @event.Type.Id);
-        if (eventSelector == null)
+        try
         {
+            return eventSelector != null && TryGetKey(eventSelector, @event, partition, out key);
+        }
+        catch (Exception ex)
+        {
+            _logger.LogWarning(ex, "Failed to get projection key for key selector type {KeySelectorType}", eventSelector.KeySelectorType);
+            key = null;
             return false;
```

Comment:
This could have happened just in the OccuredIsKey method, since all the other methods don't throw.